### PR TITLE
Make Telemetry CRD namspace scoped

### PR DESCRIPTION
**Pull request by chrkl**
## Description

The Telemetry module CRD should be namespace scoped to be aligned with other Kyma modules. This PR updates the scope and related ClusterRole. Since a cluster scoped resource cannot be owned by a namespace scoped resource, the reconciliation of the ValidatingWebhookConfiguration had to be updated.

Changes proposed in this pull request:

- Make Telemetry module CRD namespace scoped
- Update RBAC
- Change webhook reconciliation to not use owner references anymore

## Related Issues and Documents

Closes: https://github.com/kyma-project/telemetry-manager/issues/286

Related issues: https://github.com/kyma-project/busola/issues/2512

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes (env test)
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [ ] My code follows the style guidelines of this project.
- [ ] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [ ] This PR adds value and shows no feature creep.
- [ ] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -116,27 +117,9 @@ rules:
   resources:
   - telemetries
   verbs:
-  - create
-  - delete
   - get
```
Comment (rakesh-garimella) [inline config/rbac/role.yaml:120]: Do we need these to cluster scoped

Comment (chrkl) [reply config/rbac/role.yaml:120]: No, only get, list, and watch. Similar to other resources.

**Review by rakesh-garimella — DISMISSED** [review]

**Review by rakesh-garimella — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**